src/internal/EnhancedButton.js

```diff
@@ -207,21 +213,23 @@ class EnhancedButton extends Component {
     this.props.onBlur(event);
   };
 
-  handleFocus = (event) => {
-    if (event) event.persist();
-    if (!this.props.disabled && !this.props.disableKeyboardFocus) {
-      // setTimeout is needed because the focus event fires first
-      // Wait so that we can capture if this was a keyboard focus
-      // or touch focus
-      this.focusTimeout = setTimeout(() => {
-        if (tabPressed) {
-          this.setKeyboardFocus(event);
-          tabPressed = false;
-        }
-      }, 150);
-
-      this.props.onFocus(event);
-    }
+  handleClick = (event) => {
+    // when firing the handleClick from the handleKeyUp, 2 events are generated, both with the same timeStamp
+    // the chances of someone managing to fire two events in the same millisecond are ... well slim to none.
+    const bumpTime = this.state.lastHandledEventTime + 1;
+    // this is for the chance that no event is provided
+    const eventTime = event ? event.hasOwnProperty('timeStamp') ? event.timeStamp : bumpTime : bumpTime;
+    
+    // don't bother running the handler code if the event being processed happened in the same millisecond as the last one processed
+    if (eventTime !== this.state.lastHandledEventTime) {
+      this.setState({lastHandledEventTime: eventTime}); // record this for the next time so we can test if we should process the event
+      this.cancelFocusTimeout();
+      if (!this.props.disabled) {
+        tabPressed = false;
+        this.removeKeyboardFocus(event);
+        this.props.onClick(event);
+      }
+     }
```

Comment:
This function replaces `handleFocus`? Also, there are 2 `handleClick` functions. Maybe an oversight?

Reply:
The code puts back functionality dropped in the last update.  The extra code stops the same event being handled multiple times.
There should be no replacement of any function.

I'm replying on my phone right now, but if you would like a full breakdown let me know and I will go through it when I get home tonight.

Reply:
@djbuckley I'm just looking at the diff... could be a typo?

This PR replaces `handleFocus` - https://github.com/mui-org/material-ui/pull/10108/files#diff-0ab20af15832a25c3ce0468e2409753bR216

Also there is an extra `handleClick` here - https://github.com/mui-org/material-ui/pull/10108/files#diff-0ab20af15832a25c3ce0468e2409753bR235

Reply:
yep, you are right I made a bit of a cross eyed boo boo there...
I would have pulled the version in my fork but that would have introduced more issues to check.
Sorry about being so ham fisted here.